compiler_gym/envs/llvm/service/ObservationSpaces.cc

```diff
@@ -90,6 +94,70 @@ std::vector<ObservationSpace> getLlvmObservationSpaceList() {
             defaultValue.begin(), defaultValue.end()};
         break;
       }
+      case LlvmObservationSpace::IR2VEC_FA: {
+        ScalarRange featureSize;
+        featureSize.mutable_min()->set_value(0.0);
```

Comment:
I'm not sure about this. This code says that values are in the range [0,∞], but when I run your code I see plenty of negative values:

```py
>>> env.observation["Ir2vecFa"]
array([ 16.67847493,  -8.76068458, -49.90505585,  12.22742195,
        15.365804  , -10.31495722, -42.74864244, -10.26059285,
       -44.02249729,  23.81014121,  29.06372466,  24.08734658,
       -14.95525244,  19.8942756 , -29.91043964,  10.30582115,
       -24.02354845,   0.10980253,  -1.66427926,  14.17916835,
       -34.78192827,  37.14407874,  -8.33318256,  -3.45480279,
       -16.80741089, -32.45384884,  45.50566991,  37.82753753,
       -49.07060102,  -8.93597257, -52.5364784 ,   1.33546551,
       -12.41253508,  29.89899298,  10.97634208,  10.21049925,
        31.45356546,  16.61958681,  13.0980088 ,  -8.284721  ,
```

What are the bounds for embedding values?

Reply:
It seems that ScalarRange wasn't a good fit for the shape of the embeddings as the range is not really bounded. I switched this to be of a Sequence type and fixed the length of the sequence type to be of 300 for both max & min. Can you please check the new code?

Reply:
What is the shape of the two non-function-level spaces? Is it a single 300 dimension vector? Or a list of 300 dimension vectors?

Reply:
It's a single 300 dimension vector

Reply:
OKay, it should be an `int64_list` then. You can copy the Autophase space and adjust the dimensionality and limits:

https://github.com/facebookresearch/CompilerGym/blob/1553e1f553c701249719ada9eae9b2036f210e5c/compiler_gym/envs/llvm/service/ObservationSpaces.cc#L81-L94

If there is no lower bound, remove this line:

```c++
        featureSize.mutable_min()->set_value(0);
```

Reply:
> OKay, it should be an `int64_list` then. You can copy the Autophase space and adjust the dimensionality and limits:
> 
> https://github.com/facebookresearch/CompilerGym/blob/1553e1f553c701249719ada9eae9b2036f210e5c/compiler_gym/envs/llvm/service/ObservationSpaces.cc#L81-L94
> 
> If there is no lower bound, remove this line:
> 
> ```c++
>         featureSize.mutable_min()->set_value(0);
> ```

Did you mean a `double_list` ? The values for the embeddings are floating-point numbers

Reply:
Ah yes, of course, sorry :)